docs/tutorials/creating-your-first-substrate-chain/learn.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Background Information
+---
+
+In this section we will teach you about the Substrate blockchain development
+framework.
+
+## Background on Blockchains
+
+Blockchain development is hard.
+
+Blockchain networks are composed of individual **nodes** that are connected
+together on a peer-to-peer (P2P) network. Nodes are the individual computers on
+a network running the blockchain software that makes everything work.
+
+To function, a blockchain node needs:
+
+* A Database
+* P2P Networking
+* A Consensus Engine
+* Transaction Handling
+* A State Transition Function
+* and more...
+
+These technologies span a huge breadth of computer science, and thus usually
+require expert teams to develop. As a result, most blockchain projects are _not_
+developed from the ground up. Instead they are forked from already existing
+blockchains. For instance:
+
+* Bitcoin was forked to create: Litecoin, ZCash, Namecoin, Bitcoin Cash, etc...
+* Ethereum was forked to create: Quorem, POA Network, KodakCoin, Musicoin,
+  etc...
+
+![](assets/forks.png)
+
+Building blockchains in this way has serious limitations as these existing
+platform were not designed to be modified.
+
+## Substrate
+
+Substrate is an **open source**, **modular**, and **extensible** framework for
+building blockchains.
+
+Substrate has been designed from the ground up to provide a flexible framework
+for innovators to design and build their next blockchain network. It provides
+all the core components you need to build a customized blockchain node.
+
+### Substrate Node Template
+
+We provide an out-of-the-box working Substrate node in the form of the
+`substrate-node-template`, which should be compiling as you read this. Without
+making any changes, you and your friends could share this node template and
+create a working blockchain network with a cryptocurrency and everything!
+
+We will teach you how to use this node in "developer" mode, which allows you to
+run a network with a single node, and have some pre-configured user accounts
+with funds.
+
+### Substrate Runtime Module Template
+
+The Substrate framework puts an emphasis on making it easy to design custom
+block execution logic, a.k.a. the state transition function of the blockchain.
+We call this part of Substrate the **runtime**.
+
+The Substrate runtime is composed of runtime modules. You can think of these
+runtime modules as individual pieces of logic which define what your blockchain
```

Comment:
The modules don't *directly* define what the blockchain can do, but they do indirectly -- I think it'd be valuable to clarify that modules are libraries and the runtime configures these libraries into an executable. In this sense, the modules themselves do not define what the blockchain can do, but they constrain the implementation which defines what the blockchain can do.

Reply:
I have no idea the distinction you are trying to make here. The modules have logic, the logic defines the state transition function, the state transition function is the only way that the blockchain can move block to block.

Reply:
I think the distinction is that if I tell you the set of all modules in my runtime, that doesn't totally define the behavior of my runtime. You would also need to know how many instances of each module there are, how each of their configuration traits are implemented, and how their genesis configs are set up.

I'm not sure whether it's worth explaining all that at this stage or not. In general I'd prefer to let the author of the article (in _this case_ Shawn) make that call.

Reply:
Yes, this is what I was trying to communicate. The modules have logic that enables for the configuration of these modules in the runtime to define the state transition function but the logic *in* the modules doesn't directly necessarily define the state transition function...the modules determine the set of permissible state transition functions. For every finite set of modules, there is a finite number of runtime configurations, but these runtime configurations are not all the same...

Reply:
Can we take a step back and think like someone who has never used Substrate before and reading these words? What is your suggestion here?